scripts/prepare_compliance_data.py

```diff
@@ -0,0 +1,155 @@
+"""
+A script that takes the compliance dataset (the released version
+of which is at https://github.com/ga4gh/compliance/tree/master/test-data)
+and turns it into a directory bundle of binary and JSON files suitable
+for use by the reference server.
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import shutil
+import argparse
+import os
+import sys
+from os import path
+from glob import glob
+import pysam
+
+
+class ComplianceDataMunger(object):
+    def __init__(self, args):
+        self.inputDirectory = args.inputDirectory
+        self.outputDirectory = args.outputDirectory
+
+        # get all the reference files (they'll be the ones with .fa extension)
+        self.referenceFiles = map(path.basename,
+                                  glob(path.join(self.inputDirectory, "*.fa")))
+
+        self.refsetsDirectory = path.join(self.outputDirectory,
+                                          "referenceSets")
+        self.hg37Directory = path.join(self.refsetsDirectory, "hg37")
+
+        # datasets
+        self.datasetsDirectory = path.join(self.outputDirectory, "datasets")
+        self.readFiles = map(path.basename,
+                             (glob(path.join(self.inputDirectory, "*.sam"))))
+        self.variantFiles = map(path.basename,
+                                (glob(path.join(self.inputDirectory,
+                                                "*.vcf"))))
+
+        self.datasets = [d for d in set([p.split('_')[0] for p in
+                                         self.readFiles + self.variantFiles])]
+        self.datasetReads = dict()
+        self.datasetVariants = dict()
+        for ds in self.datasets:
+            self.datasetReads[ds] = [r for r in
+                                     self.readFiles if r.startswith(ds)]
+
+        # Variants themselves are split into groups,
+        # based on second part of the _ split:
+        for ds in self.datasets:
+            self.datasetVariants[ds] = dict()
+            # only those variants inside this dataset
+            dsvlist = [v for v in self.variantFiles if v.startswith(ds)]
+            # create nested dictionary based on group belonging
+            for dsv in dsvlist:
+                dsvGroup = dsv.split('_')[1]
+                self.datasetVariants[ds][dsvGroup] = \
+                    self.datasetVariants[ds].get(dsvGroup, []) + [dsv]
+
+        self.datasetDirs = [path.join(self.outputDirectory, ds)
+                            for ds in self.datasets]
+
+    def run(self):
+        if not path.exists(self.outputDirectory):
+            os.makedirs(self.outputDirectory)
+
+        # Clean out, make and re-populate references directory
+        # For now, assume a single, statically-named referenceSet
+        print("Converting references...", file=sys.stderr)
```

Comment:
Normally with unix programs, verbose messages go to stderr, with
stdout being used for piping data.  Note that
logging.handlers.StreamHandler defaults to stderr.

Although messages like this are not that useful.  If it needs to
talk to me, I would rather have it tell me after that fact that
it important so many records, just to give a sense that it did
something.



Danny Colligan <notifications@github.com> writes:
> In scripts/prepare_compliance_data.py:
> 
> > +            # create nested dictionary based on group belonging
> > +            for dsv in dsvlist:
> > +                dsvGroup = dsv.split('_')[1]
> > +                self.datasetVariants[ds][dsvGroup] = \
> > +                    self.datasetVariants[ds].get(dsvGroup, []) + [dsv]
> > +
> > +        self.datasetDirs = [path.join(self.outputDirectory, ds)
> > +                            for ds in self.datasets]
> > +
> > +    def run(self):
> > +        if not path.exists(self.outputDirectory):
> > +            os.makedirs(self.outputDirectory)
> > +
> > +        # Clean out, make and re-populate references directory
> > +        # For now, assume a single, statically-named referenceSet
> > +        print("Converting references...", file=sys.stderr)
> 
> why does this go to sys.stderr instead of sys.stdout?
> 
> —
> Reply to this email directly or view it on GitHub.*
>